azkaban-exec-server/src/main/java/azkaban/container/FlowContainer.java

```diff
@@ -0,0 +1,336 @@
+    int execId = 0;
+    try {
+      execId = Integer.parseInt(execIdStr);
+    } catch (NumberFormatException ne) {
+      System.out.printf("Execution ID %s is invalid %n", args[0]);
+    }
+    System.out.printf("Execution ID = %d%n", execId);
+
+    // Setup work directories
+    final Path currentWorkingDir = Paths.get("").toAbsolutePath();
+    final Path projectDirPath = Paths.get(currentWorkingDir.toString(), PROJECT_DIR);
+    final Path jobtypePluginPath = Paths.get(currentWorkingDir.toString(), JOBTYPE_DIR);
+
+    // Move files to respective dirs
+    // Create project dir
+    // TODO: Need to revisit
+    System.out.println("Creating project dir");
+    Files.createDirectory(projectDirPath);
+    Path projectZipPath = Paths.get(projectDirPath.toString(), projectZipName);
+    System.out.println("moving projectDir:" + projectZipName + ": to " +
+        projectZipPath);
+    Files.move(Paths.get(projectZipName), projectZipPath);
+    // TODO : revisit this logic,
+    // Unzip the project zip
+    FlowContainer.unzipFile(projectZipPath.toString(), projectDirPath);
+
+    // TODO : Deepak - throw away after directory setup is final.
+    Path pwd = Paths.get("").toAbsolutePath();
+    System.out.println("Deepak : Dumping dir structure for = " + pwd.toString());
+    java.nio.file.Files.walk(pwd).filter(java.nio.file.Files::isRegularFile)
+        .forEach(System.out::println);
+
+    // Set Azkaban props
+    Props props = setAzkabanProps(jobtypePluginPath);
+
+    // Inject AzkabanCommonModule
+    final Injector injector = Guice.createInjector(
+        new AzkabanCommonModule(props)
+    );
+    SERVICE_PROVIDER.setInjector(injector);
+
+    // Constructor
+    final FlowContainer flowContainer =
+        new FlowContainer(projectDirPath, props);
+
+    // Use some execId to execute the flow
+    flowContainer.submitFlow(execId);
+  }
+
+  // Set Azkaban Props
+  private static Props setAzkabanProps(final Path jobtypePluginPath) {
+    final Map<String, String> propsMap = new HashMap<>();
+    propsMap.put(AzkabanExecutorServer.JOBTYPE_PLUGIN_DIR,
+        jobtypePluginPath.toString());
+
+    // Setup the azkaban.properties here.
+    final String[] args = {CONF_ARG, CONF_DIR};
+    final Props props = AzkabanServer.loadProps(args);
+
+    // Set db stuff.
+    return new Props(props, propsMap);
+  }
+
+  // Submit flow
+  public void submitFlow(final int execId) throws ExecutorManagerException {
+    final FlowRunner flowRunner = createFlowRunner(execId);
+    submitFlowRunner(flowRunner);
+  }
+
+  // create Flow Runner
+  private FlowRunner createFlowRunner(final int execId) throws ExecutorManagerException {
+    final ExecutableFlow flow = this.executorLoader.fetchExecutableFlow(execId);
+    if (flow == null) {
+      throw new ExecutorManagerException("Error loading flow with exec " + execId);
+    }
+
+    // Update the status of the flow from DISPATCHING to PREPARING
+    flow.setStatus(Status.PREPARING);
```

Comment:
I think setting the status to PREPARING should happen right in the beginning of the entry point into FlowContainer. Doing it at this stage skips a couple of early steps and hence, may provide a more confusing picture as to time spent between DISPATCHING vs. PREPARING stage

Reply:
+1. PREPARING is to capture the time taken before flow is actually ready to launch (first job can be started.). Status PREPARING should be set in the beginning. It also needs to be saved in database. Here is not yet saved in database.

Reply:
The status can only be set when flow object is available as it is a flow attribute. The status is set first thing after the flow object is created.

Reply:
Flow Object can be created in main method once you verify that execId is appropriate. Please set status over there.

Reply:
I still think it is at the correct place.

Reply:
It is not. Flow should be in preparing state when FlowContainer starts. It should be in running state when flowrunner starts running.